### PR TITLE
Upcast Celu for it to support more dtypes | feat(torchlib)

### PR DESCRIPTION
**Pull request by justinchuby**
Celu only supports float in ONNX. This change adds the variant to support more floating types by manually adding casts in the function.

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/nn.py [hunk 1]
```diff
@@ -347,6 +350,14 @@ def aten_celu(self: FLOAT, alpha: float = 1.0) -> FLOAT:
     return op.Celu(self, alpha=alpha)  # op.Celu only support float32
 
 
+@torch_op("aten::celu")
+def aten_celu_upcasted(self: TFloatUnlessFloat32, alpha: float = 1.0) -> TFloatUnlessFloat32:
```
Comment (BowenBao) [inline onnxscript/function_libs/torch_lib/ops/nn.py:354]: nit: hmm we are also downcasting from double. I know I started with the term "upcast" in my original issue, but maybe we should name it something else to callout the onnx dtype limitation? `aten_celu_dtype_hack`? Just an idea and we might need to come up with something more formal if there are more cases.

**Review by BowenBao — APPROVED** [review]

File: onnxscript/function_libs/torch_lib/ops/nn.py (hunk 1 above)
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/nn.py:354]:
```suggestion
def aten_celu_type_promoted(self: TFloatUnlessFloat32, alpha: float = 1.0) -> TFloatUnlessFloat32:
```

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/nn.py:354]: Updating to type_promoted. wdyt?

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 2]
```diff
@@ -1059,6 +1059,7 @@ def _where_input_wrangler(
         reason="fixme: RuntimeError: ORT inference error GlobalAveragePool. https://github.com/microsoft/onnxruntime/issues/16449",
     ),
     TorchLibOpInfo("nn.functional.celu", nn_ops.aten_celu),
+    TorchLibOpInfo("nn.functional.celu_upcasted", nn_ops.aten_celu_upcasted),
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:1062]:
```suggestion
    TorchLibOpInfo("nn.functional.celu_type_promoted", nn_ops.aten_celu_type_promoted),
```

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 3]
```diff
@@ -2126,6 +2127,11 @@ def _where_input_wrangler(
     "nn.functional.scaled_dot_product_attention",
     ("nn.functional.scaled_dot_product_attention_bool_mask",),
 )
+ops_test_common.duplicate_opinfo(
+    OPS_DB,
+    "nn.functional.celu",
+    ("nn.functional.celu_upcasted",),
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:2133]:
```suggestion
    ("nn.functional.celu_type_promoted",),
```

**Pull request merged**